exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SchemaUtils.kt

```diff
@@ -624,6 +666,72 @@ object SchemaUtils {
         return toCreate.toList()
     }
 
+    /**
+     * @param scriptName The name to be used for the generated migration script.
+     * @param scriptDirectory The directory in which to create the migration script.
+     * @param withLogs By default, a description for each intermediate step, as well as its execution time, is logged at
+     * the INFO level. This can be disabled by setting [withLogs] to `false`.
+     *
+     * @return The generated migration script.
+     *
+     * This function simply generates the migration script without applying the migration. The purpose of it is to show
+     * the user what the migration script will look like before applying the migration.
+     */
+    @ExperimentalDatabaseMigrationApi
+    fun generateMigrationScript(vararg tables: Table, scriptDirectory: String, scriptName: String, withLogs: Boolean = true): File {
+        val allStatements = statementsRequiredForDatabaseMigration(*tables, withLogs = withLogs)
+
+        val migrationScript = File("$scriptDirectory/$scriptName.sql")
+        val migrationScriptExists = if (migrationScript.exists()) true else migrationScript.createNewFile()
+        if (migrationScriptExists) {
+            // Clear existing content
+            migrationScript.writeText("")
```

Comment:
Should the KDocs include a warning that the content of an existing script will be overwritten?